workflow/executor/executor.go

```diff
@@ -720,28 +744,53 @@ func (we *WorkflowExecutor) AnnotateOutputs(ctx context.Context, logArt *wfv1.Ar
 	if !outputs.HasOutputs() {
 		return nil
 	}
-	log.Infof("Annotating pod with output")
-	outputBytes, err := json.Marshal(outputs)
+
+	if err := we.annotatePodWithOutputs(ctx, outputs); !apierr.IsForbidden(err) { // me were either successful (nil) or some other error
+		return err
+	}
+
+	return we.updateTaskSetStatusWithOutputs(ctx, outputs)
+}
+
+func (we *WorkflowExecutor) updateTaskSetStatusWithOutputs(ctx context.Context, outputs *wfv1.Outputs) error {
+	log.Info("Patching taskset status with outputs")
+	tasksetInterface := we.workflowInterface.ArgoprojV1alpha1().WorkflowTaskSets(we.Namespace)
+	return waitutil.Backoff(ExecutorRetry, func() (bool, error) {
+		taskset, err := tasksetInterface.Get(ctx, we.workflowName, metav1.GetOptions{})
+		if err != nil && !errorsutil.IsTransientErr(err) {
+			return false, err
+		}
+		if taskset.Status == nil {
```

Comment:
repeated logic here in a hard to test place, instead can we add a `taskSet.SetNodeResult()` func?